alphagov/government-frontend
# Migrate to Dart Sass from LibSass

## What

Migrate to Dart Sass from LibSass.

## Why

- LibSass is now [deprecated](https://sass-lang.com/blog/libsass-is-deprecated/)
- Making changes ahead of Design System team dropping support for LibSass https://github.com/alphagov/govuk-frontend/discussions/2632.

## Visual Changes

None.

## How can I test these changes?

- run `yarn jasmine:prepare` (or run `bundle exec rake app:assets:clobber app:assets:precompile`)
- what should happen? The Sass files should be compiled. Keep an eye on the command line for warnings indicating the compilation process using Dart Sass
```
Deprecation Warning: Using / for division outside of calc() is deprecated and will be removed in Dart Sass 2.0.0.
Recommendation: math.div(300%, 4) or calc(300% / 4)
More info and automated migrator: https://sass-lang.com/d/slash-div
   │
26 │   three-quarters: (300% / 4),
   │                    ^^^^^^^^
   ...
```
- the `/builds` folder: `app/assets/builds` should now be populated with CSS files. Note, the 
- subsequently, all assets ought to be generated successfully within the `/public` directory
- no test failures to indicate the correct number of stylesheets are generated
- deploy to Integration. You'll notice that in the '**build and publish image**' task, the steps taken resemble those mentioned earlier, `dartsass-rails` is bundled and any Dart Sass warnings encountered during the compilation process are reported)
- inspect several pages, e.g. [
  guide](https://government-frontend-pr-3034.herokuapp.com/log-in-register-hmrc-online-services), [detailed guide](https://government-frontend-pr-3034.herokuapp.com/guidance/waste-exemption-nwfd-2-temporary-storage-at-the-place-of-production--2), [HTML publication](https://government-frontend-pr-3034.herokuapp.com/government/publications/budget-2016-documents/budget-2016), [news article](https://government-frontend-pr-3034.herokuapp.com/government/news/the-personal-independence-payment-amendment-regulations-2017-statement-by-paul-gray), [specialist document](https://government-frontend-pr-3034.herokuapp.com/business-finance-support/access-to-finance-advice-north-west-england), [statistical dataset](https://government-frontend-pr-3034.herokuapp.com/government/statistical-data-sets/unclaimed-estates-list), [foreign travel advice](https://government-frontend-pr-3034.herokuapp.com/foreign-travel-advice/nepal), [worldwide organisation](https://government-frontend-pr-3034.herokuapp.com/world/organisations/british-embassy-madrid) to ensure the absence of errors.

## Anything else

Once the PR gets the green light, any temporary commits and references to local gems e.g. `gem "govuk_publishing_components", path: "../govuk_publishing_components"` will be removed.

See:
- https://github.com/alphagov/collections/pull/3447
- https://github.com/alphagov/govuk_publishing_components/pull/3726
- https://github.com/alphagov/static/pull/3190
- https://github.com/alphagov/govuk-developer-docs/pull/4216.
